utils/build-script-impl

```diff
@@ -565,21 +573,32 @@ function set_deployment_target_based_options() {
                 -DSWIFT_DARWIN_DEPLOYMENT_VERSION_WATCHOS="${DARWIN_DEPLOYMENT_VERSION_WATCHOS}"
                 -DLLVM_ENABLE_LIBCXX:BOOL=TRUE
             )
-
-            if [[ "${llvm_target_arch}" ]] ; then
-                llvm_cmake_options=(
-                    "${llvm_cmake_options[@]}"
-                    -DLLVM_TARGET_ARCH="${llvm_target_arch}"
-                )
-            fi
-
             ;;
         *)
-            echo "Unknown compiler deployment target: ${deployment_target}"
+            echo "Unknown host tools target: ${host}"
             exit 1
             ;;
     esac
 
+    if [[ "${llvm_target_arch}" ]] ; then
+        llvm_cmake_options+=(
+            -DLLVM_TARGET_ARCH="${llvm_target_arch}"
+        )
+    fi
+    if [[ "${llvm_host_triple}" ]] ; then
+        llvm_cmake_options+=(
+            -DLLVM_HOST_TRIPLE:STRING="${llvm_host_triple}"
+        )
+    fi
+    if [[ "${SWIFT_HOST_VARIANT}" ]] ; then
+        swift_cmake_options+=(
+            -DSWIFT_HOST_VARIANT="${SWIFT_HOST_VARIANT}"
+        )
+    fi
+    swift_cmake_options+=(
+        -DSWIFT_HOST_VARIANT_SDK="${SWIFT_HOST_VARIANT_SDK}"
+        -DSWIFT_HOST_VARIANT_ARCH="${SWIFT_HOST_VARIANT_ARCH}"
```

Comment:
You're right, I've got another patch which cleans up that part of the CMake files. It uses `SWIFT_HOST_VARIANT_SDK` and `SWIFT_HOST_VARIANT_ARCH` to calculate the Swift compiler host/stdlib targets (instead of directly parsing the `CMAKE_SYSTEM_*` variables like we do today), but falls back to parsing from the CMAKE_SYSTEM if they're not set.

I could add it to this PR or do it separately; It's self-contained.